src/serializer/serializer.js

```diff
@@ -134,6 +134,13 @@ export class Serializer {
     if (this.logger.hasErrors()) return undefined;
     if (timingStats !== undefined) timingStats.deepTraversalTime = Date.now() - timingStats.deepTraversalTime;
 
+    const realmPreludeGenerator = this.realm.preludeGenerator;
+    invariant(realmPreludeGenerator);
+    const residualHeapValueIdentifiers = new ResidualHeapValueIdentifiers(
+      residualHeapVisitor.values.keys(),
+      realmPreludeGenerator.getValueNameGenerator()
```

Comment:
This seems to be the only remaining place where the value-name-generator is actually referenced? If so, I suggest to encapsulate it entirely in the `ResidualHeapValueIdentifiers` class, and not expose it via the `PreludeGenerator` --- it's not necessary to expose it, and by encapsulating it we could prevent later bad uses of it.